### ISSUE TITLE
[CI/CD] build-package (deb) is broken

### ISSUE THREAD

**Issue by songy23** [open-telemetry/opentelemetry-collector-contrib#31443]
### Describe the issue you're reporting

See recent failures https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8065610224/job/22032135180?pr=31421, https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8065600280/job/22032028700

```
# test install
echo
podman run --name $container_name -d $image_name

Error: OCI runtime error: chmod `run/shm`: Operation not supported
podman rm -fv otelcontribcol-deb-test >/dev/null 2>&1 || true
Error: Process completed with exit code 126.
```

**Comment by crobert-1** [comment]
Looks like this is https://github.com/containers/crun/issues/1308. The workaround for now may have to be manually downloading the fixed `crun` package and updating the path for podman. [Workaround reference.](https://noobient.com/2023/11/15/fixing-ubuntu-containers-failing-to-start-with-systemd/)

**Comment by cwegener** [comment]
Oh. Awesome. My change to podman worked and errors are now easier to resolve by everyone. :smile: 

Interesting how this error only started occurring recently ... must have been a change in Kernel version in the GH runner image?

This also reminds me of #30239 I'll put a PR together for it.

**Comment by cwegener** [comment]
Quick root cause: the runner image kernel upgrade from `6.2.0-1019-azure` to  `6.5.0-1015-azure` likely introduced the regression (likely because of the regression from linux 6.6 being backported to 6.5 and included in the 6.5 azure kernel from Canonical/Microsoft). https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20240225.1

And yeah, the crun fix from the containers community probably was not possible to be backported to the old crun codebase that ships with ubuntu 22.04? https://github.com/containers/crun/commit/be16ee75ff8574698250352302e9d5496d888d69

I think this should be a bug ticket with the `actions/runner-images` repo.
